source/patterns/@aws-solutions-constructs/core/lib/input-validation.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+import * as kinesis from '@aws-cdk/aws-kinesis';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as mediastore from '@aws-cdk/aws-mediastore';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as sns from '@aws-cdk/aws-sns';
+import * as glue from '@aws-cdk/aws-glue';
+import * as sagemaker from '@aws-cdk/aws-sagemaker';
+import * as secretsmanager from "@aws-cdk/aws-secretsmanager";
+
+export interface VerifiedProps {
+  dynamoTableProps?: dynamodb.TableProps,
+  existingTableObj?: dynamodb.Table,
+
+  existingStreamObj?: kinesis.Stream;
+  kinesisStreamProps?: kinesis.StreamProps,
+
+  existingLambdaObj?: lambda.Function,
+  lambdaFunctionProps?: lambda.FunctionProps,
+
+  existingQueueObj?: sqs.Queue,
+  queueProps?: sqs.QueueProps,
+  deployDeadLetterQueue?: boolean,
+  deadLetterQueueProps?: sqs.QueueProps,
+
+  existingMediaStoreContainerObj?: mediastore.CfnContainer;
+  mediaStoreContainerProps?: mediastore.CfnContainerProps;
+
+  existingBucketObj?: s3.Bucket,
+  bucketProps?: s3.BucketProps,
+
+  // topicsProps is an incorrect attribute used in event-rule-sns that 
+  // we need to support
+  topicProps?: sns.TopicProps,
+  topicsProps?: sns.TopicProps,
+  existingTopicObj?: sns.Topic,  
+
+  glueJobProps?: glue.CfnJobProps,
+  existingGlueJob?: glue.CfnJob,
+
+  existingSagemakerEndpointObj?: sagemaker.CfnEndpoint,
+  endpointProps?: sagemaker.CfnEndpointProps,
+
+  readonly existingSecretObj?: secretsmanager.Secret;
+  readonly secretProps?: secretsmanager.SecretProps;
+
+}
+
+export function CheckProps(propsObject: VerifiedProps | any) {
+  let errorMessages = '';
+  let errorFound = false;
+
+  if (propsObject.dynamoTableProps && propsObject.existingTableObj) {
+    errorMessages += 'Cannot specify an existing DDB table AND DDB table props\n';
```

Comment:
Make the error message more user friendly by saying "Either provide `existingTableObj` or `dynamoTableProps`, not both", similar feedback for the rest of error messages